api/types/resource.go

```diff
@@ -504,13 +504,8 @@ func (m *Metadata) CheckAndSetDefaults() error {
 // MatchLabels takes a map of labels and returns `true` if the resource has ALL
 // of them.
 func MatchLabels(resource ResourceWithLabels, labels map[string]string) bool {
-	if len(labels) == 0 {
-		return true
-	}
-
-	resourceLabels := resource.GetAllLabels()
-	for name, value := range labels {
-		if resourceLabels[name] != value {
+	for key, value := range labels {
+		if v, ok := resource.GetLabel(key); !ok || v != value {
```

Comment:
looking over the various implementations of `GetAllLabels` and `GetLabel` most look consistent, but I did notice that `AppServerOrSAMLIdPServiceProviderV1` has an impl that differs slightly (failing to search one of the possible label locations in `GetLabel`).  Probably worth fixing that.